### ISSUE TITLE
Please update to Web IDL changes to optional dictionary defaulting

### ISSUE THREAD

**Issue by bzbarsky** [WebAudio/web-audio-api#1989]
Web IDL recently changed to require explicitly listing a default value for defaulted dictionaries instead of it just magically happening.

In practice what that means is that APIs like this:
```
 Constructor (BaseAudioContext context, optional AnalyserOptions options)
```
need to be written like this instead:
```
 Constructor (BaseAudioContext context, optional AnalyserOptions options = {})
```
(with the addition of = {} on the dictionary argument). This affects a bunch of the web audio APIs.

**Comment by padenot** [comment]
Taking this, this is just a bunch of manual fixes but probably rather straightforward.

**Comment by padenot** [comment]
Fixed by #1993, thanks for the heads up.
